src/hayhooks/server/utils/deploy_utils.py

```diff
@@ -279,11 +285,29 @@ async def run_endpoint_without_files(run_req: request_model) -> response_model:
 def add_pipeline_api_route(app: FastAPI, pipeline_name: str, pipeline_wrapper: BasePipelineWrapper) -> None:
     clog = log.bind(pipeline_name=pipeline_name)
 
-    docstring = docstring_parser.parse(inspect.getdoc(pipeline_wrapper.run_api) or "")
-    RunRequest = create_request_model_from_callable(pipeline_wrapper.run_api, f'{pipeline_name}Run', docstring)
-    RunResponse = create_response_model_from_callable(pipeline_wrapper.run_api, f'{pipeline_name}Run', docstring)
-
-    run_api_params = inspect.signature(pipeline_wrapper.run_api).parameters
+    # Determine which run_api method to use (prefer async if available)
+    if pipeline_wrapper._is_run_api_async_implemented:
+        run_method_to_inspect = pipeline_wrapper.run_api_async
+        clog.debug("Using `run_api_async` as API route handler.")
+    elif pipeline_wrapper._is_run_api_implemented:
+        run_method_to_inspect = pipeline_wrapper.run_api
+        clog.debug("Using `run_api` as API route handler.")
+    else:
+        # If neither run_api nor run_api_async is implemented,
+        # this pipeline will not have a generic /<pipeline_name>/run endpoint.
+        # This is a valid configuration (e.g., for chat-only pipelines).
+        clog.warning(
+            f"Pipeline '{pipeline_name}' does not implement `run_api` or `run_api_async`. "
+            f"Skipping /{pipeline_name}/run API route creation."
+        )
+        return
+
+    docstring_content = inspect.getdoc(run_method_to_inspect) or ""
+    docstring = docstring_parser.parse(docstring_content)
+    RunRequest = create_request_model_from_callable(run_method_to_inspect, f'{pipeline_name}Run', docstring)
```

Comment:
Comment that this dynamic pydantic model generation? 🙏

Reply:
It is actually written in `create_request_model_from_callable` docstring. Do you think it will be more clear adding another comment here?

Reply:
Ah ok then! All good